Doc/library/argparse.rst

```diff
@@ -11,6 +11,18 @@
 
 **Source code:** :source:`Lib/argparse.py`
 
+.. note::
+
+   While :mod:`argparse` is the recommended standard library module for
```

Comment:
I would not recommend it for now. You can use it on your own risk. `optparse` may need more work, but it works as expected. I would recommend it for beginners, which cannot distinguish their own errors from peculiarities of the library.

Reply:
For this PR, I'd like to stick with the status quo as far as recommendations go (that is, only taking the step back from "`argparse` is the only non-deprecation argument processing option in the standard library").

Taking the extra step to saying "argparse is *not* recommended over optparse anymore, even for end user applications" would then be a separate follow up question.